addons/mail/static/src/core/common/message_search_hook.js

```diff
@@ -0,0 +1,90 @@
+/* @odoo-module */
+
+import { useSequential } from "@mail/utils/common/hooks";
+import { useState, onWillUnmount, markup } from "@odoo/owl";
+import { useService } from "@web/core/utils/hooks";
+
+/**
+ * @param {string} searchTerm
+ * @param {string} target
+ */
+export function searchHighlight(searchTerm, target) {
+    if (!searchTerm) {
+        return target;
+    }
+    const htmlDoc = new DOMParser().parseFromString(target, "text/html");
+    for (const term of searchTerm.split(" ")) {
+        const matchs = htmlDoc.evaluate(
+            `//*[text()[contains(translate(., '${term.toUpperCase()}', '${term.toLowerCase()}'), '${term.toLowerCase()}')]]`,
+            htmlDoc,
+            null,
+            XPathResult.ORDERED_NODE_SNAPSHOT_TYPE
+        );
+        for (let i = 0; i < matchs.snapshotLength; i++) {
+            const element = matchs.snapshotItem(i);
+            const newNode = [];
+            for (const node of element.childNodes) {
+                if (node.nodeType === Node.TEXT_NODE) {
+                    const span = htmlDoc.createElement("span");
+                    span.innerHTML = node.textContent.replaceAll(
```

Comment:
This is not correct, `textContent` cannot be inserted back as `innerHTML` as it would effectively unescape everything that was escaped before.

You can test `<strong>a</strong> test` and search for anything matching, it will display `a test` (with bold), instead it should display back `<strong>a</strong> test` (with proper highlighting).

There should be a test for that.